src/FffGcodeWriter.cpp

```diff
@@ -90,18 +63,54 @@ void FffGcodeWriter::writeGCode(SliceDataStorage& storage, TimeKeeper& time_keep
             mesh_order_per_extruder.push_back(calculateMeshOrder(storage, extruder_nr));
         }
     }
+    calculateExtruderOrderPerLayer(storage);
 
-    for(unsigned int layer_nr=0; layer_nr<total_layers; layer_nr++)
+    int process_layer_starting_layer_nr = 0;
+    bool has_raft = getSettingAsPlatformAdhesion("adhesion_type") == EPlatformAdhesion::RAFT;
+    if (has_raft)
     {
-        processLayer(storage, layer_nr, total_layers);
+        processRaft(storage, total_layers);
+        // process filler layers to fill the airgap with helper object (support etc) so that they stick better to the raft.
+        process_layer_starting_layer_nr = -Raft::getFillerLayerCount(storage);
     }
-    
+
+
+    const std::function<LayerPlan* (int)>& produce_item =
+        [&storage, total_layers, this](int layer_nr)
+        {
+            LayerPlan& gcode_layer =processLayer(storage, layer_nr, total_layers);
+            return &gcode_layer;
+        };
+    const std::function<void (LayerPlan*)>& consume_item =
+        [this](LayerPlan* gcode_layer)
+        {
+            layer_plan_buffer.push(*gcode_layer);
+            LayerPlan* to_be_written = layer_plan_buffer.processBuffer();
+            if (to_be_written)
+            {
+                to_be_written->writeGCode(gcode);
+                delete to_be_written;
+            }
+        };
+    const unsigned int max_task_count = 20; // TODO: hardcoded value
```

Comment:
So... we going to do this then? We could at least make it a `#DEFINE` or something static.